### PR TITLE
OP-398 increase coverage in pregtreattype; format Manager file

### PR DESCRIPTION
**Pull request by dbmalkovsky**
Coverage before:
image

Coverage after:
image

### CODE REVIEW

File: src/test/java/org/isf/pregtreattype/test/Tests.java [hunk 1]
```diff
@@ -116,6 +121,141 @@ public void testIoDeletePregnantTreatmentType() throws Exception {
...
 	}
 
+	@Test
+	public void testMgrGetPregnantTreatmentType() throws Exception {
+		String code = _setupTestPregnantTreatmentType(false);
+		PregnantTreatmentType foundPregnantTreatmentType = pregnantTreatmentTypeIoOperationRepository.findOne(code);
+		ArrayList<PregnantTreatmentType> pregnantTreatmentTypes = pregnantTreatmentTypeBrowserManager.getPregnantTreatmentType();
+
+		for (PregnantTreatmentType pregnantTreatmentType : pregnantTreatmentTypes) {
+			if (pregnantTreatmentType.getCode().equals(code)) {
```
Comment (emecas) [inline src/test/java/org/isf/pregtreattype/test/Tests.java:131]: What about asserting something for the else branch?

File: src/test/java/org/isf/pregtreattype/test/Tests.java [hunk 2]
```diff
@@ -116,6 +121,141 @@ public void testIoDeletePregnantTreatmentType() throws Exception {
 		assertThat(result).isFalse();
 	}
 
+	@Test
+	public void testMgrGetPregnantTreatmentType() throws Exception {
+		String code = _setupTestPregnantTreatmentType(false);
+		PregnantTreatmentType foundPregnantTreatmentType = pregnantTreatmentTypeIoOperationRepository.findOne(code);
+		ArrayList<PregnantTreatmentType> pregnantTreatmentTypes = pregnantTreatmentTypeBrowserManager.getPregnantTreatmentType();
```
Comment (emecas) [inline src/test/java/org/isf/pregtreattype/test/Tests.java:128]: if you setup more PregnantTreatmentType elements, then you could assert that the method under tests getPregnantTreatmentType() returns a list of the expected size

**Review by emecas — APPROVED** [review]
Thanks @dbmalkovsky 

```
Results :

Tests run: 675, Failures: 0, Errors: 0, Skipped: 2

[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:29 min
[INFO] Finished at: 2021-01-11T17:20:14-05:00
[INFO] ------------------------------------------------------------------------
```

File: src/test/java/org/isf/pregtreattype/test/Tests.java (hunk 1 above)
Comment (dbmalkovsky) [reply src/test/java/org/isf/pregtreattype/test/Tests.java:137]: I added something but I'm not sure how much value it adds.

File: src/test/java/org/isf/pregtreattype/test/Tests.java (hunk 2 above)
Comment (dbmalkovsky) [reply src/test/java/org/isf/pregtreattype/test/Tests.java:133]: I created two more instances and did a `hasSize(3)` assertion.  Thanks

File: src/test/java/org/isf/pregtreattype/test/Tests.java (hunk 1 above)
Comment (mwithi) [reply src/test/java/org/isf/pregtreattype/test/Tests.java:137]: @emecas what was you thinking like?

**Review by mwithi — APPROVED** [review]
A part adding more assertions, it looks good for me too.
```
Results :

Tests run: 939, Failures: 0, Errors: 0, Skipped: 1
```

File: src/test/java/org/isf/pregtreattype/test/Tests.java (hunk 1 above)
Comment (mwithi) [reply src/test/java/org/isf/pregtreattype/test/Tests.java:137]: @emecas +1

Comment (emecas) [reply src/test/java/org/isf/pregtreattype/test/Tests.java:137]: Initially I just noticed that branch was not being  tested, and was a good spot for exploratory testing.

`isNotEqualTo` makes more sense because new version of the test  has multiple `PregnantTreatmentType `objects in the ArrayList

Also, it totally depends how you write your tests,  and at this point I would think manager implementation is pretty robust, but as I mentioned before, exploratory testing and adding edge conditions could uncover early/hidden issues and shield your logic for changes. 

For example, just in order to develop a little better what I'm trying to express here, with a test like the following, and keeping the focus on testing `getPregnantTreatmentType()` method,  you would catch issues in equals method, the  absent of a index in the data model allowing duplicate records, problems about the order of the expected results, etc:

```
	@Test
	public void testMgrGetPregnantTreatmentType() throws Exception {
		String code = _setupTestPregnantTreatmentType(false);
		String code2 = _setupTestPregnantTreatmentType(false);
		PregnantTreatmentType foundPregnantTreatmentType = pregnantTreatmentTypeIoOperationRepository.findOne(code);
		PregnantTreatmentType pregnantTreatmentType2 = new PregnantTreatmentType("AA", "AA description");
		pregnantTreatmentTypeIoOperationRepository.saveAndFlush(pregnantTreatmentType2);
		PregnantTreatmentType pregnantTreatmentType3 = new PregnantTreatmentType("BB", "BB description");
		pregnantTreatmentTypeIoOperationRepository.saveAndFlush(pregnantTreatmentType3);
		
		PregnantTreatmentType pregnantTreatmentType3b = new PregnantTreatmentType("BB", "BB description");
		pregnantTreatmentTypeIoOperationRepository.saveAndFlush(pregnantTreatmentType3b);

		ArrayList<PregnantTreatmentType> pregnantTreatmentTypes = pregnantTreatmentTypeBrowserManager.getPregnantTreatmentType();
		assertThat(pregnantTreatmentTypes).hasSize(3);
		
		PregnantTreatmentType pregnantTreatmentType4 = new PregnantTreatmentType("AA1", "BB description");
		pregnantTreatmentTypeIoOperationRepository.saveAndFlush(pregnantTreatmentType4);
		PregnantTreatmentType pregnantTreatmentType5 = new PregnantTreatmentType("BB1", "AA description");
		pregnantTreatmentTypeIoOperationRepository.saveAndFlush(pregnantTreatmentType5);
		
		pregnantTreatmentTypes = pregnantTreatmentTypeBrowserManager.getPregnantTreatmentType();
		assertThat(pregnantTreatmentTypes).hasSize(5);
		
		
		String description = "AA";
		
		for (PregnantTreatmentType pregnantTreatmentType : pregnantTreatmentTypes) {
			if (pregnantTreatmentType.getCode().equals(code)) {
				assertThat(pregnantTreatmentType.getDescription()).isEqualTo(foundPregnantTreatmentType.getDescription());
			} else {
				assertThat(pregnantTreatmentType.getDescription()).isNotEqualTo(foundPregnantTreatmentType.getDescription());
			}
			
			assertThat(description).isLessThanOrEqualTo(pregnantTreatmentType.getDescription());
			description = pregnantTreatmentType.getDescription();
		}
	}
```

Comment (emecas) [reply src/test/java/org/isf/pregtreattype/test/Tests.java:137]: BTW this is already resolved!!

**Review by emecas — APPROVED** [review]

File: src/test/java/org/isf/pregtreattype/test/Tests.java (hunk 1 above)
Comment (mwithi) [reply src/test/java/org/isf/pregtreattype/test/Tests.java:137]: Thank you guys!
